### PR TITLE
3id connect & metamask

### PR DESCRIPTION
**Pull request by uberbinge**

### CODE REVIEW

File: frontend/src/contexts/3idConnect.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { EthereumAuthProvider, ThreeIdConnect } from '@3id/connect'
+import ThreeIdResolver from '@ceramicnetwork/3id-did-resolver'
+import Ceramic from '@ceramicnetwork/http-client'
+import { IDX } from '@ceramicstudio/idx'
+import { DID } from 'dids'
+
+import { web3Modal } from './providers'
+
+const CERAMIC_URL = process.env.CERAMIC_API || 'http://localhost:7007'
```
Comment (elmariachi111) [inline frontend/src/contexts/3idConnect.ts]: REACT_APP_CERAMIC_URL

File: frontend/src/contexts/3idConnect.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+const threeIdConnect = new ThreeIdConnect()
+
+const authenticate = async () => {
+  const ethProvider = await web3Modal.connect()
+  const addresses = await ethProvider.enable()
+
+  const authProvider = new EthereumAuthProvider(ethProvider, addresses[0])
```
Comment (elmariachi111) [inline frontend/src/contexts/3idConnect.ts:17]: That code must go into the Ceramic context

File: frontend/src/contexts/3idConnect.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    resolver: ThreeIdResolver.getResolver(ceramic)
+  })
+
+  await did.authenticate()
+  console.log(did.id)
+
+  const jws = await did.createJWS({ hello: 'world' })
+  console.log(jws)
+
+  window.idx = new IDX({ ceramic })
```
Comment (elmariachi111) [inline frontend/src/contexts/3idConnect.ts]: see how to set those in Ceramic context (Ceramic.tsx)

File: frontend/src/contexts/Providers.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import WalletConnectProvider from '@walletconnect/web3-provider'
+import Authereum from 'authereum'
+import Fortmatic from 'fortmatic'
+import Web3Modal from 'web3modal'
+
+const providerOptions = {
+  portis: {
+    package: Portis,
+    options: {
+      id: '8f5cf962-ad62-4861-ab0c-7b234b6e6cff'
```
Comment (elmariachi111) [inline frontend/src/contexts/Providers.ts:11]: Don't put private secrets in git :( -> move them to .env and *after* the hackathon, rotate them on your account

File: frontend/src/contexts/Providers.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+  },
+  authereum: {
+    package: Authereum,
+    options: {}
+  }
+}
+
+export const web3Modal = new Web3Modal({
+  network: 'mainnet',
```
Comment (mohamedmansour) [inline frontend/src/contexts/Providers.ts:33]: @elmariachi111  this should be dynamic? right? Why do we need Web3Modal?

File: frontend/src/contexts/Providers.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  portis: {
+    package: Portis,
+    options: {
+      id: '8f5cf962-ad62-4861-ab0c-7b234b6e6cff'
+    }
+  },
+  walletconnect: {
+    package: WalletConnectProvider,
+    options: {
+      infuraId: 'e87f83fb85bf4aa09bdf6605ebe144b7'
```
Comment (mohamedmansour) [inline frontend/src/contexts/Providers.ts:17]: @elmariachi111  @waah42 move this to environment variable like ceramic

File: frontend/src/contexts/3idConnect.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  window.ceramic = ceramic
+  window.did = did.id
+}
+
+document.getElementById('bauth').addEventListener('click', authenticate)
```
Comment (mohamedmansour) [inline frontend/src/contexts/3idConnect.ts]: What is this for?

File: frontend/src/contexts/Providers.ts (hunk 4 above)
Comment (uberbinge) [reply frontend/src/contexts/Providers.ts:11]: it's not my secret got it from 3idconnect repo.

File: frontend/src/contexts/Providers.ts (hunk 5 above)
Comment (uberbinge) [reply frontend/src/contexts/Providers.ts:33]: not sure.. i am following https://github.com/ceramicstudio/3id-connect/tree/develop/apps/example

File: frontend/src/contexts/3idConnect.ts (hunk 7 above)
Comment (uberbinge) [reply frontend/src/contexts/3idConnect.ts]: removed

**Pull request closed without merging**